specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/mfe.json

```diff
@@ -0,0 +1,5889 @@
+        },
+        {
+          "type": "object",
+          "properties": {
+            "jobType": {
+              "enum": [
+                "Command",
+                "Sweep",
+                "Labeling",
+                "Pipeline",
+                "Data",
+                "AutoML"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "JobType",
+                "modelAsString": true
+              }
+            },
+            "status": {
+              "description": "Status of the job.",
+              "enum": [
+                "NotStarted",
+                "Starting",
+                "Provisioning",
+                "Preparing",
+                "Queued",
+                "Running",
+                "Finalizing",
+                "CancelRequested",
+                "Completed",
+                "Failed",
+                "Canceled",
+                "NotResponding",
+                "Paused"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "JobStatus",
+                "modelAsString": true
+              }
+            },
+            "maxRunDurationSeconds": {
+              "format": "int64",
+              "description": "The max run duration in seconds, after which the job will be cancelled.",
+              "type": "integer"
+            },
+            "codeConfiguration": {
+              "description": "Code configuration of the job.",
+              "allOf": [
+                {
+                  "$ref": "#/definitions/CodeConfiguration"
+                }
+              ]
+            },
+            "environmentId": {
+              "description": "Environment specification of the job.",
+              "type": "string"
+            },
+            "dataBindings": {
+              "description": "Mapping of data bindings used in the job.",
+              "type": "object",
+              "additionalProperties": {
+                "$ref": "#/definitions/DataBinding"
+              }
+            },
+            "distributionConfiguration": {
+              "$ref": "#/definitions/DistributionConfiguration"
+            },
+            "parameters": {
+              "description": "Input parameters.",
+              "type": "object",
+              "additionalProperties": {}
+            }
+          }
+        }
+      ],
+      "discriminator": "jobType",
+      "x-ms-discriminator-value": "Command",
+      "additionalProperties": false
+    },
+    "ParameterSamplingConfiguration": {
+      "description": "class for all hyperparameter sampling algorithms",
+      "required": [
+        "parameterSpace",
+        "samplingType"
+      ],
+      "type": "object",
+      "properties": {
+        "parameterSpace": {
+          "description": "A dictionary containing each parameter and its distribution. The dictionary key is the name of the parameter",
+          "type": "object"
```

Comment:
Is this definition for parameterSpace acceptable?

Reply:
If this is as tightly as you can define it, sure.  If all the values will be a certain type you can specify that using additionalProperties